src/Compilers/CSharp/Portable/BoundTree/BoundDecisionDag.cs

```diff
@@ -180,6 +180,12 @@ BoundDecisionDagNode makeReplacement(BoundDecisionDagNode dag, Func<BoundDecisio
                             return d.Value == inputConstant;
                         case BoundDagTypeTest d:
                             return inputConstant.IsNull ? (bool?)false : null;
+                        case BoundDagRelationalTest d:
+                            var f = ValueSetFactory.ForSpecialType(input.Type.SpecialType);
+                            if (f is null) return null;
+                            // TODO: When ValueSetFactory has a method for comparing two values, use it.
```

Comment:
>TODO [](start = 31, length = 4)

PROTOTYPE?

Reply:
No.  The code works correctly as is.  This is a suggestion for a future optimization, which I expect to happen after merging into master.

---
In reply to: [395854439](https://github.com/dotnet/roslyn/pull/42501#discussion_r395854439) [](ancestors = 395854439)

Reply:
Should this be also filed as a follow-up issue?

Reply:
That wouldn't be particularly helpful to me, though I don't object to it.

---
In reply to: [397252372](https://github.com/dotnet/roslyn/pull/42501#discussion_r397252372) [](ancestors = 397252372)